### PR TITLE
Added more description to hall_parameter signature in parameters.py

### PR DESCRIPTION
**Pull request by Dhawal-Modi**
For issue #915 
I have added some docstrings for `hall_parameter`. Copied `particle` docstring from existing docstring of `kappa_thermal_speed`. .Wasn't sure with what to do with `ion` and `particle`. Also added `hall_parameter` equation.

### CODE REVIEW

**Comment by Tiger-Du** [comment]
Here are my thoughts:

1) I think that the parameters of `ion` and `particle` are confusing because `particle` could be a type of ion (as the function is implemented now) and `ion` is a type of particle. Perhaps a better naming scheme would be "field particles" and "test particles" to replace `ion` and `particle` respectively, as is done in the docstrings of `fundamental_electron_collision_freq` and  `fundamental_ion_collision_freq`. However, from what I understand, for the contexts in plasma physics in which the Hall parameter is important, the test particles are the electrons of the plasma, and the field particles are the ions of the plasma. So perhaps it makes sense to just call `particle` "electron".
2) The two functions that compute `coll_rate` both assume a "slowly moving Maxwellian distribution." I think that this assumption should be included in the first line of the docstring of `Hall_parameter.`
3) If V is not specified as a parameter to `Hall_parameter,` then the two functions that compute `coll_rate` both assume that V is thermal velocity. I think that this sentence, "If not provided, thermal velocity is assumed: :math:`\mu V^2 \sim 2 k_B T` where `mu` is the reduced mass.", should be added to the end of the description of V under Parameters in the docstring.

**Comment by Tiger-Du** [comment]
Another naming scheme is "test particle" and "target particle", as is done in the docstring of the function named `mean_free_path` in `plasmapy/formulary/collisions.py`. I think that "field particles" and "test particles" is more understandable, however, so perhaps the docstring of `mean_free_path` should be changed. But either way the naming scheme should be the same.

File: plasmapy/formulary/parameters.py [hunk 1]
```diff
@@ -827,6 +830,14 @@ def Hall_parameter(
...
 
+    Notes
+    -----
+    The Hall Parameter is the ratio between the gyrofrequency and the ion-particle
+    collision rate(also known as collision frequency)
+
+    .. math::
+        \beta = \frac{\omega_c}{\nu}
+
     Examples
```
Comment (namurphy) [inline plasmapy/formulary/parameters.py:841]:
```suggestion
    .. math::
        \beta = \frac{\omega_c}{\nu}

    where :math:`\omega_c` is the particle's gyrofrequency and :math:`\nu` is
    the collision frequency between ``particle`` and ``ion``.

    Examples
```

**Review by namurphy — DISMISSED** [review]
This is helpful.  Thank you!  I suggested a few slight changes, and it's worth thinking about @Tiger-Du's suggestions, either here or for a follow-up pull request.

**Comment by namurphy** [comment]
> 1. I think that the parameters of `ion` and `particle` are confusing because `particle` could be a type of ion (as the function is implemented now) and `ion` is a type of particle. Perhaps a better naming scheme would be "field particles" and "test particles" to replace `ion` and `particle` respectively, as is done in the docstrings of `fundamental_electron_collision_freq` and  `fundamental_ion_collision_freq` in `plasmapy/formulary/collisions.py`. However, from what I understand, for the contexts in plasma physics in which the Hall parameter is important, the test particles are the electrons of the plasma, and the field particles are the ions of the plasma. So perhaps it makes sense to just call `particle` "electron".

Yeah, I like your ideas for `test_particle` and either `target_particle` or `field_particle` for things like this.  You're right that the test particles are usually electrons, though I did find a couple of cases (e.g., this [thesis](https://alfven.princeton.edu/publications/thesis/sankaran-thesis-2005)) where both the electron Hall parameter and the ion Hall parameter were used.  I believe `ion` does need to be an ion, so a possibility would be `target_ion` or `field_ion`.  That would be a breaking change so I'd suggest a different pull request.

> 2. The two functions that compute `coll_rate` both assume a "slowly moving Maxwellian distribution". I think that this assumption should be included in the first line of the docstring of `Hall_parameter`.

Yep, we should mention that.  I'd suggest in the `Notes` section.

> 3. If `V` is not specified as a parameter to `Hall_parameter`, then the two functions that compute `coll_rate` both assume that `V` is thermal velocity. I think that this sentence, "If not provided, thermal velocity is assumed: :math:`` `\mu V^2 \sim 2 k_B T` `` where `` `mu` `` is the reduced mass.", should be added to the end of the description of `V` under "Parameters" in the docstring.

Good point!

**Comment by Dhawal-Modi** [comment]
Okay let me also implement the suggestions put forward by Tiger-Du!

**Review by Dhawal-Modi — COMMENTED** [review]
Changes done as requested.

File: changelog/934.doc.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Added description to Hall_parameter signature and equation in docstrings.
```
Comment (rocco8773) [inline changelog/934.doc.rst:1]:
```suggestion
Added description to :func:`~plasmapy.formulary.parameters.Hall_parameter`
signature and equation in docstrings.
```

The will allow the `Hall_parameter` to be automatically linked in the changelog.

File: plasmapy/formulary/parameters.py [hunk 3]
```diff
@@ -801,15 +801,18 @@ def Hall_parameter(
     Parameters
```
Comment (rocco8773) [inline plasmapy/formulary/parameters.py:801]: I felt there was still a lot to be done to clarify what is being calculated and how the function's arguments link to the Hall parameter expression, so I tried to make this more explicit.  I also updated links between referenced functionality.  So, here's my take at what the docstring should look like...

```python
r"""
Calculate the ``particle`` Hall parameter for a plasma.

The Hall parameter for plasma species :math:`s` (``particle``) is given by:

.. math::

    \beta_{s} = \frac{\Omega_{c s}}{\nu_{s s^{\prime}}}

where :math:`\Omega_{c s}` is the gyrofrequncy for plasma species :math:`s`
(``particle``) and :math:`\nu_{s s^{\prime}}` is the collison frequency
between plasma species :math:`s` (``particle``) and species
:math:`s^{\prime}` (``ion``).

**Aliases:** `betaH_`

Parameters
----------
n : `~astropy.units.quantity.Quantity`
    The number density associated with ``particle``.
T : `~astropy.units.quantity.Quantity`
    The temperature of associated with ``particle``.
B : `~astropy.units.quantity.Quantity`
    The magnetic field.
ion : `~plasmapy.particles.Particle`
    The type of ion ``particle`` is colliding.
particle : `~plasmapy.particles.Particle`
    The particle species for which the Hall parameter is calculated for.
    Representation of the particle species (e.g., 'p' for protons, 'D+' for
    deuterium, or 'He-4 +1' for singly ionized helium-4). If no charge state
    information is provided, then the particles are assumed to be singly
    charged.
coulomb_log : `float`, optional
    Preset value for the Coulomb logarithm. Used mostly for testing purposes.
V : `~astropy.units.quantity.Quantity`
    The relative velocity between ``particle`` and ``ion``.  If not provided,
    then the ``particle`` thermal velocity is assumed
    (`~plasmapy.formulary.parameters.thermal_speed`).
coulomb_log_method : `str`, optional
    Method used for Coulomb logarithm calculation. (see
    `~plasmapy.formulary.collisions.Coulomb_logarithm`)

See Also
--------
plasmapy.formulary.parameters.gyrofrequency
plasmapy.formulary.collisions.fundamental_electron_collision_freq
plasmapy.formulary.collisions.fundamental_ion_collision_freq
plasmapy.formulary.collisions.Coulomb_logarithm

Returns
-------
`~astropy.units.quantity.Quantity`
    Hall parameter for ``particle``.

Notes
-----
* For calculating the collision frequency,
  `~plamsapy.formulary.colisions.fundamental_electron_collision_freq` is used
  when ``particle`` is an electron and
  `~plamsapy.formulary.colisions.fundamental_ion_collision_freq` when
  ``particle`` is an ion.
* The collision frequencies are calculated assuming a slowly moving
  Maxwellian distribution.

Examples
--------
>>> from astropy import units as u
>>> Hall_parameter(1e10 * u.m**-3, 2.8e3 * u.eV, 2.3 * u.T, 'He-4 +1', 'e-')
<Quantity 7.26446...e+16>
>>> Hall_parameter(1e10 * u.m**-3, 5.8e3 * u.eV, 2.3 * u.T, 'He-4 +1', 'e-')
<Quantity 2.11158...e+17>
"""
```

**Review by rocco8773 — CHANGES_REQUESTED** [review]
Thank you for taking this up!  I like the changes you made, but thought the relationship between the function's arguments and the Hall parameter expression could be more explicit.  So, my suggestions focus around that.

I also added in some of the suggestions made by @Tiger-Du in https://github.com/PlasmaPy/PlasmaPy/pull/934#issuecomment-733512592.

I agree with @Tiger-Du that the ``ion`` and ``particle`` nomenclature is not very clear.  I did not change any of that here, since it would break backwards compatibility and thought the more explicit references I suggested for the docsrings could compensate for this issue.

@namurphy In the future, I think this function should be updated so a user could pass in their own collision frequency and not just rely on the ones we calculate.  Thoughts? (Obviously for a future PR, and we can updated ``particle`` vs ``ion` at the same time.)

File: changelog/934.doc.rst (hunk 2 above)
Comment (namurphy) [reply changelog/934.doc.rst:1]: Oh!  I didn't know you could do this.  Good to know!

**Review by Dhawal-Modi — COMMENTED** [review]
Okay so i'm a little bit confused now as i have not visited this PR for a long time now. Have the changes  requested reflect on the actual commits? I am new to Open Source Contributions so might need a little help.

**Comment by rocco8773** [comment]
@Dhawal-Modi Sorry I haven't been able to give you review comments sooner, my time has been consumed for the past few weeks with another large project.  For many of us PlasmaPy is not the only project we are working on, so unfortunately we can be a little slow with reviews at times.  Your PR is nearly complete, so we should be able to merge soon...

With respect to my comments, you do NOT need to open a new PR.  That was a side discussion I was having with @namurphy about another "unrelated" issue to open with respect to the `Hall_parameter`.  I apologize for my code dump in my review https://github.com/PlasmaPy/PlasmaPy/pull/934#discussion_r541990310, but I'm only allowed to make direct comments on lines that you first changed and I had comments related to the entire docstring.  If you make the changes I suggested in https://github.com/PlasmaPy/PlasmaPy/pull/934#discussion_r541990310, then I'd be happy to merge this PR.

File: plasmapy/formulary/parameters.py (hunk 3 above)
Comment (rocco8773) [reply plasmapy/formulary/parameters.py:801]: My suggested docstring will render like this in the documentation...

image

File: plasmapy/formulary/parameters.py [hunk 4]
```diff
@@ -791,41 +791,69 @@ def Hall_parameter(
...
-        The particle of interest.
-    coulomb_log : float, optional
+    n : `~astropy.units.quantity.Quantity`
+        The number density associated with ``particle``.
+    T : `~astropy.units.quantity.Quantity`
+        The temperature of associated with ``particle``.
+    B : `~astropy.units.quantity.Quantity`
+        The magnetic field.
+    ion : `~plasmapy.particles.Particle`
+        The type of ion ``particle`` is colliding.
```
Comment (rocco8773) [inline plasmapy/formulary/parameters.py:819]:
```suggestion
        The type of ion ``particle`` is colliding with.
```

File: plasmapy/formulary/parameters.py [hunk 5]
```diff
@@ -791,41 +791,69 @@ def Hall_parameter(
...
+    particle : `~plasmapy.particles.Particle`
+        The particle species for which the Hall parameter is calculated for.
+        Representation of the particle species (e.g., 'p' for protons, 'D+' for
+        deuterium, or 'He-4 +1' for singly ionized helium-4). If no charge state
+        information is provided, then the particles are assumed to be singly
+        charged.
```
Comment (rocco8773) [inline plasmapy/formulary/parameters.py:825]:
```suggestion
        The particle species for which the Hall parameter is calculated for.
        Representation of the particle species (e.g., ``'p'`` for protons,
        ``'D+'`` for deuterium, or ``'He-4 +1'`` for singly ionized helium-4).
        If no charge state information is provided, then the particles are
        assumed to be singly charged.
```

File: plasmapy/formulary/parameters.py [hunk 6]
```diff
@@ -791,41 +791,69 @@ def Hall_parameter(
...
 
     Returns
     -------
-    astropy.units.quantity.Quantity
+    `~astropy.units.quantity.Quantity`
+        Hall parameter for ``particle``.
+
+    Notes
+    -----
+    * For calculating the collision frequency,
```
Comment (rocco8773) [inline plasmapy/formulary/parameters.py:850]:
```suggestion
    * For calculating the collision frequency
```

**Review by rocco8773 — CHANGES_REQUESTED** [review]
Sorry, just a few more minor fixes.

**Review by Dhawal-Modi — COMMENTED** [review]
Sorry have been busy with work couldn't get around to it sooner.

**Review by rocco8773 — APPROVED** [review]

**Comment by rocco8773** [comment]
No worries, that is life.

This looks good so I'll go ahead an merge.  Thank you for taking this on!

**Pull request merged**